pandas/io/excel/_xlsxwriter.py

```diff
@@ -235,3 +235,51 @@ def write_cells(
                 )
             else:
                 wks.write(startrow + cell.row, startcol + cell.col, val, style)
+
+    def write_table(
```

Comment:
I don't think this needs to be a separate method at all. Can't you just use the normal write method and apply the table after cells are written out?

Reply:
I think that can be done, but there are several differences with the normal write function so don't think the result will be that nice. Key differences:

 - merge cells are not allowed for tables
 - header needs to be dealt with differently
 - individual call styling makes little sense (don't think it should be supported or at least discouraged) for tables

Reply:
Disregard the above, I think I can make it work with not too big adjustments